include/onnxruntime/core/session/onnxruntime_c_api.h

```diff
@@ -537,6 +537,11 @@ typedef struct OrtOpenVINOProviderOptions {
 struct OrtApi;
 typedef struct OrtApi OrtApi;
 
+#ifdef ENABLE_TRAINING_ON_DEVICE
```

Comment:
are users expected to define this symbol before including this file? what if there's a mismatch with how ORT was built?

Reply:
With the new set of changes, we don't expect the users to define this symbol. If that package built is not for training, the `GetTrainingApi` will return a `nullptr`. Will push the commit soon.